test/python/test_instructions.py

```diff
@@ -270,6 +271,100 @@ def test_get_all_hwm(
         f"Expected test hmw {test_hwm} but got {received_test_hwm}"
 
 
+def build_preattestation(op_level, op_round, chain_id):
+    """Build a preattestation."""
+    return Preattestation(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_attestation(op_level, op_round, chain_id):
+    """Build a attestation."""
+    return Attestation(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_attestation_dal(op_level, op_round, chain_id):
+    """Build a attestation_dal."""
+    return AttestationDal(
+        op_level=op_level,
+        op_round=op_round
+    ).forge(chain_id=chain_id)
+
+def build_block(level, current_round, chain_id):
+    """Build a block."""
+    return Block(
+        header=BlockHeader(
+            level=level,
+            fitness=Fitness(current_round=current_round)
+        )
+    ).forge(chain_id=chain_id)
+
+PARAMETERS = [
+    (build_attestation,    (0, 0), build_preattestation,  (0, 1), True ),
+    (build_block,          (0, 1), build_attestation_dal, (1, 0), True ),
+    (build_block,          (0, 1), build_attestation_dal, (0, 0), False),
+    (build_attestation,    (1, 0), build_preattestation,  (0, 1), False),
+] + [
+    (build_preattestation,  (0, 0), build_preattestation,  (0, 0), False),
+    (build_preattestation,  (0, 0), build_block,           (0, 0), False),
+    (build_preattestation,  (0, 0), build_attestation,     (0, 0), True ),
+    (build_preattestation,  (0, 0), build_attestation_dal, (0, 0), True ),
+    (build_attestation,     (0, 0), build_preattestation,  (0, 0), False),
+    (build_attestation,     (0, 0), build_attestation,     (0, 0), False),
+    (build_attestation,     (0, 0), build_attestation_dal, (0, 0), False),
+    (build_attestation,     (0, 0), build_block,           (0, 0), False),
+    (build_attestation_dal, (0, 0), build_preattestation,  (0, 0), False),
+    (build_attestation_dal, (0, 0), build_attestation,     (0, 0), False),
+    (build_attestation_dal, (0, 0), build_attestation_dal, (0, 0), False),
+    (build_attestation_dal, (0, 0), build_block,           (0, 0), False),
+    (build_block,           (1, 1), build_preattestation,  (1, 1), True ),
+    (build_block,           (1, 1), build_attestation,     (1, 1), True ),
+    (build_block,           (1, 1), build_attestation_dal, (1, 1), True ),
+    (build_block,           (1, 1), build_block,           (1, 1), False),
```

Comment:
I was unable to sign a block at reset level (with round 0).
Is this behaviour expected?

Reply:
There can not be a 0 block. Minimum level is 1. See `baking_auth.c:17`
`bool is_valid_level(level_t lvl) {
    return !(lvl & 0xC0000000);
}`

Reply:
I am not sure to understand:
`is_valid_level(0) == is_valid_level(1) == true`
This `is_valid_level` is only used to check that the level does not exceed `2^30`.

Reply:
Its unusual if you are not able to sign block with 0,0 combination for level, round. I dont see any  check in the code regarding this constraint.

Reply:
I think it's because at start (or after a reset), the `level`, in the `HWM`, is 0 (or the one set) and the `round` is 0.
So, trying to sign a block at this `level` and this `round` leads, in the function `is_level_authorized`, to:
 - `baking_info->level == hwm->highest_level`
 - `baking_info->round == hwm->highest_round`
 - `baking_info->type == BAKING_TYPE_TENDERBAKE_BLOCK`
 
And this is not authorized
Whereas with `baking_info->type == BAKING_TYPE_TENDERBAKE_(PRE/)ENDORSEMENT`, there are some cases authorized.

Reply:
is there a way we can make it explicit, either in a comment or writing an if condition ?

Reply:
Yes !
I can write a test for it and add a comment.
But I'm not sure whether this is an expected behaviour.

Reply:
According to tezos economic protocol , the genesis block is still level 0. But it contains dictator key to activate protocol. As for how this relates to ledger, let us brainstorm in our Monday meeting. 
1. Does this limitation have any effect on using ledger on testnet or sandboxed environment?
2. Can we resolve it without affecting double baking restriction?

Reply:
I open an issue: #22